core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -617,9 +617,10 @@ class GroupCoordinator(val brokerId: Int,
                        leavingMembers: List[MemberIdentity],
                        responseCallback: LeaveGroupResult => Unit): Unit = {
 
-    def removeCurrentMemberFromGroup(group: GroupMetadata, memberId: String): Unit = {
+    def removeCurrentMemberFromGroup(group: GroupMetadata, memberId: String, reason: Option[String]): Unit = {
       val member = group.get(memberId)
-      removeMemberAndUpdateGroup(group, member, s"Removing member $memberId on LeaveGroup")
+      val leaveReason = reason.getOrElse("unknown reason")
+      removeMemberAndUpdateGroup(group, member, s"Removing member $memberId on LeaveGroup due to: $leaveReason")
```

Comment:
I think that we should use the same pattern for both join reasons and leave reason. See https://github.com/apache/kafka/pull/11566#discussion_r771371068.